CppCoreGuidelines.md

```diff
@@ -10077,14 +10077,18 @@ If at all possible, reduce the conditions to a simple set of alternatives (e.g.,
 
 ##### Example
 
-    bool owned = false;
-    owner<istream*> inp = [&]{
+    unique_ptr<istream> owner;
+    auto& in = [&] {
         switch (source) {
-        case default:       owned = false; return &cin;
-        case command_line:  owned = true;  return new istringstream{argv[2]};
-        case file:          owned = true;  return new ifstream{argv[2]};
+            case command_line:  owner = new istringstream{argv[2]}; return *owner;
+            case file:          owner = new ifstream{argv[2]}; return *owner;
```

Comment:
I can restore it if you like, but I also have read Code Complete :), which advises against it. 
 And there's also http://stackoverflow.com/a/101984/6610.

Shall I put each statement on a separate line?

    case command_line:
       owner = new istringstream{argv[2]};
       return *owner;
...